### PR TITLE
Various Things From patching_todo.txt

### PR DESCRIPTION
**Pull request by Steviegt6**
### What is the new feature?
Did various things from patching_todo.txt.
Check the individual commits for more detail.

### CODE REVIEW

**Comment by Steviegt6** [comment]
Would still be smart to double-check and make sure all cases of 22 used for checks relating to maximum buffs have been replaced with MaxBuffs.

File: patches/tModLoader/Terraria/Player.cs.patch [hunk 1]
```diff
@@ -1864,7 +1926,7 @@
  					GamepadEnableGrappleCooldown();
  				}
 -				else if (Main.tile[myX, myY].type == 4 || Main.tile[myX, myY].type == 13 || (Main.tile[myX, myY].type == 50 && Main.tile[myX, myY].frameX == 90)) {
-+				else if (TileLoader.IsTorch(Main.tile[myX, myY].type) || Main.tile[myX, myY].type == 13 || (Main.tile[myX, myY].type == 50 && Main.tile[myX, myY].frameX == 90)) {
++				else if (TileLoader.IsTorch(Main.tile[myX, myY].type) || Main.tile[myX, myY].type == 13 || (Main.tile[myX, myY].type == 50 && Main.tile[myX, myY].frameX == 90) || TileID.Sets.CanDropFromRightClick[Main.tile[myX, myY].type] == true) {
```
Comment (Mirsario) [inline patches/tModLoader/Terraria/Player.cs.patch:1929]: Real "If bool == true" moment

File: patches/tModLoader/Terraria/ID/TileID.cs.patch [hunk 2]
```diff
@@ -32,6 +32,14 @@
  			public static bool[] Corrupt = Factory.CreateBoolSet(23, 25, 112, 163, 398, 400);
  			public static bool[] Hallow = Factory.CreateBoolSet(109, 117, 116, 164, 402, 403, 115);
  			public static bool[] Crimson = Factory.CreateBoolSet(199, 203, 234, 200, 399, 401, 205);
+@@ -195,6 +_,7 @@
+ 			public static bool[] IgnoredInHouseScore = Factory.CreateBoolSet(4, 3, 73, 82, 83, 84);
+ 			public static bool[] SpreadOverground = Factory.CreateBoolSet(2, 23, 32, 60, 70, 109, 199, 352, 477, 492);
+ 			public static bool[] SpreadUnderground = Factory.CreateBoolSet(23, 109, 199, 60, 70);
++			public static bool[] CanDropFromRightClick = Factory.CreateBoolSet(4);
```
Comment (Mirsario) [inline patches/tModLoader/Terraria/ID/TileID.cs.patch:39]: Can there be partial nested classes? If so, this should be in one. FYI, the convention for partials is now X.TML.cs

**Review by Mirsario — CHANGES_REQUESTED** [review]
Good help.

File: patches/tModLoader/Terraria/Player.cs.patch (hunk 1 above)
Comment (Steviegt6) [reply patches/tModLoader/Terraria/Player.cs.patch:1929]: lmao oops

**Review by Mirsario — APPROVED** [review]

**Comment by Mirsario** [comment]
As I accept, I notice that this is targeting the wrong branch. mergedtesting is totally a throwaway, target 1.4.

**Comment by Steviegt6** [comment]
Moved to https://github.com/tModLoader/tModLoader/pull/1089

**Pull request closed without merging**